node/node.go

```diff
@@ -701,3 +708,73 @@ func (node *Node) ShutDown() {
 	fmt.Print(msg)
 	os.Exit(0)
 }
+
+func (node *Node) populateSelfAddresses(epoch *big.Int) {
+	// reset the self addresses
+	node.SelfAddresses = map[string]common.Address{}
```

Comment:
this may not be thread safe to do. It's reset when the call uses a different epoch, which can be problematic.